### PR TITLE
Langchain SambaStudio Chat Model  integration 

### PR DESCRIPTION
**Pull request by snova-jorgep**

### CODE REVIEW

File: utils/model_wrappers/api_gateway.py [hunk 1]
```diff
@@ -213,26 +213,45 @@ def load_llm(
...
+        do_sample: Optional[bool] = False,
+        process_prompt: Optional[bool] = False,
+        stream_options: Optional[Dict[str, bool]] = {"include_usage": True},
+        special_tokens: Optional[Dict[str, str]] = {
+            'start': '<|begin_of_text|>',
+            'start_role': '<|begin_of_text|><|start_header_id|>{role}<|end_header_id|>',
+            'end_role': '<|eot_id|>',
+            'end': '<|start_header_id|>assistant<|end_header_id|>\n',
+        },
+        model_kwargs: Optional[Dict[str, Any]] = None,
```
Comment (snova-jorgep) [inline utils/model_wrappers/api_gateway.py:232]: missing sambastudio_url y sambastudio_api_key

File: utils/model_wrappers/api_gateway.py [hunk 2]
```diff
@@ -246,14 +265,37 @@ def load_chat(
...
+                do_sample=do_sample,
+                process_prompt=process_prompt,
+                stream_options=stream_options,
+                special_tokens=special_tokens,
+                model_kwargs=model_kwargs
+
+            )
+
+        elif type == 'sncloud':
+            model = ChatSambaNovaCloud(
```
Comment (snova-jorgep) [inline utils/model_wrappers/api_gateway.py:287]: envs = {
            'sambanova_url': sambanova_url,
            'sambanova_api_key': sambanova_api_key,
        }
        envs = {k: v for k, v in envs.items() if v is not None}
        
        
     should inside each model conditional

Comment (snova-jorgep) [reply utils/model_wrappers/api_gateway.py:287]: @snova-luiss

File: utils/model_wrappers/api_gateway.py (hunk 1 above)
Comment (snova-jorgep) [reply utils/model_wrappers/api_gateway.py:232]: @snova-luiss

**Review by snova-kwasia — APPROVED** [review]

**Pull request merged**